src/FilterAdder.js

```diff
@@ -1,23 +1,31 @@
 import _ from 'lodash/fp'
 import React from 'react'
-import { contexturify } from './utils/hoc'
+import { observer } from 'mobx-react'
+import Picker from './layout/NestedPicker'
+import { defaultTheme, withOptionalNode, withLoader } from './utils/hoc'
 import { newNodeFromField } from './utils/search'
 
 export let fieldsToOptions = _.map(x => ({ value: x.field, ...x }))
 
 let getGroupFields = node => _.map('field', _.getOr([], 'children', node))
 
-let FilterAdder = ({ tree, node, path, fields, Picker, uniqueFields }) => {
+let FilterAdder = _.flow(
+  observer,
+  defaultTheme({ Picker }),
+  withOptionalNode,
```

Comment:
In this specific case, the node should truly be optional (it's only used to do filtering for `uniqueFields`), so we use this flow instead of the normal `contexturify`